### ISSUE TITLE
<Spell ('Lantern of Power')> attempted a same-zone move in <Zone.SETASIDE: 6>

### ISSUE THREAD

**Issue by Ragowit** [jleclanche/fireplace#325]
I have been trying to catch up with my AI fork, and there's been a lot of work on Fireplace since last time I did anything. The implementation of Discover is one of these things. So it might just be that have done something wrong, but this happened for me.

These are my last lines from the debug log.

[fireplace.actions]: one plays <Minion ('Arch-Thief Rafaam')> (target=None, index=None)
[fireplace.entity]: one pays 9 mana
[fireplace.card]: <Minion ('Arch-Thief Rafaam')> moves from <Zone.HAND: 3> to <Zone.PLAY: 1>
[fireplace.actions]: <Minion ('Arch-Thief Rafaam')> triggering <TargetedAction: Battlecry(TARGET=None)> targeting [<Minion ('Arch-Thief Rafaam')>]
[fireplace.actions]: Activating <Minion ('Arch-Thief Rafaam')> action targeting None
[fireplace.actions]: <Minion ('Arch-Thief Rafaam')> triggering <TargetedAction: Discover(CARDS=RandomID({}))> targeting [Player(name='one', hero=<Hero ('Jaina Proudmoore')>)]
[fireplace.actions]: <Minion ('Arch-Thief Rafaam')> discovers [<Spell ('Lantern of Power')>, <Spell ('Timepiece of Horror')>, <Spell ('Mirror of Doom')>] for one
[fireplace.card]: <Spell ('Lantern of Power')> attempted a same-zone move in <Zone.SETASIDE: 6>
[fireplace.card]: <Spell ('Timepiece of Horror')> attempted a same-zone move in <Zone.SETASIDE: 6>
[fireplace.card]: <Spell ('Mirror of Doom')> attempted a same-zone move in <Zone.SETASIDE: 6>
[fireplace.entity]: Empty stack, refreshing auras and processing deaths
[fireplace.entity]: Empty stack, refreshing auras and processing deaths

Entire debug log: http://pastebin.com/DnJJEHRQ

**Comment by jleclanche** [comment]
Yeah - that's harmless actually. I could avoid it but I don't think there's any need at this point.
